packages/graphql/src/schema-model/annotation/Annotation.ts

```diff
@@ -17,140 +17,119 @@
+import type { KeyAnnotation } from "./KeyAnnotation";
+import type { AuthenticationAnnotation } from "./AuthenticationAnnotation";
+import type { TimestampAnnotation } from "./TimestampAnnotation";
+import type { FilterableAnnotation } from "./FilterableAnnotation";
+import type { JWTClaimAnnotation } from "./JWTClaimAnnotation";
+import type { QueryAnnotation } from "./QueryAnnotation";
+import type { CoalesceAnnotation } from "./CoalesceAnnotation";
+import type { SubscriptionsAuthorizationAnnotation } from "./SubscriptionsAuthorizationAnnotation";
+import type { MutationAnnotation } from "./MutationAnnotation";
+import type { UniqueAnnotation } from "./UniqueAnnotation";
 
-export enum AnnotationsKey {
-    authentication = "authentication",
-    authorization = "authorization",
-    coalesce = "coalesce",
-    customResolver = "customResolver",
-    cypher = "cypher",
-    default = "default",
-    filterable = "filterable",
-    fulltext = "fulltext",
-    id = "id",
-    jwtClaim = "jwtClaim",
-    jwtPayload = "jwtPayload",
-    key = "key",
-    limit = "limit",
-    mutation = "mutation",
-    plural = "plural",
-    populatedBy = "populatedBy",
-    private = "private",
-    query = "query",
-    selectable = "selectable",
-    settable = "settable",
-    subscription = "subscription",
-    subscriptionsAuthorization = "subscriptionsAuthorization",
-    timestamp = "timestamp",
-    unique = "unique",
-    relayId = "relayId",
+export interface Annotation {
+    readonly name: string;
 }
 
-export type Annotations = {
-    [AnnotationsKey.cypher]: CypherAnnotation;
-    [AnnotationsKey.authorization]: AuthorizationAnnotation;
-    [AnnotationsKey.authentication]: AuthenticationAnnotation;
-    [AnnotationsKey.key]: KeyAnnotation;
-    [AnnotationsKey.subscriptionsAuthorization]: SubscriptionsAuthorizationAnnotation;
-    [AnnotationsKey.limit]: LimitAnnotation;
-    [AnnotationsKey.default]: DefaultAnnotation;
-    [AnnotationsKey.coalesce]: CoalesceAnnotation;
-    [AnnotationsKey.customResolver]: CustomResolverAnnotation;
-    [AnnotationsKey.id]: IDAnnotation;
-    [AnnotationsKey.mutation]: MutationAnnotation;
-    [AnnotationsKey.plural]: PluralAnnotation;
-    [AnnotationsKey.filterable]: FilterableAnnotation;
-    [AnnotationsKey.fulltext]: FullTextAnnotation;
-    [AnnotationsKey.populatedBy]: PopulatedByAnnotation;
-    [AnnotationsKey.query]: QueryAnnotation;
-    [AnnotationsKey.private]: PrivateAnnotation;
-    [AnnotationsKey.selectable]: SelectableAnnotation;
-    [AnnotationsKey.settable]: SettableAnnotation;
-    [AnnotationsKey.timestamp]: TimestampAnnotation;
-    [AnnotationsKey.unique]: UniqueAnnotation;
-    [AnnotationsKey.subscription]: SubscriptionAnnotation;
-    [AnnotationsKey.jwtClaim]: JWTClaimAnnotation;
-    [AnnotationsKey.jwtPayload]: JWTPayloadAnnotation;
-    [AnnotationsKey.relayId]: RelayIDAnnotation;
+type CheckAnnotationName<T> = {
+    [P in keyof T]: T[P] & { name: P };
 };
```

Comment:
While I'm still not 100% happy with complex typings, I can see the benefit of adding this one, maybe adding a quick comment just so anyone reading this have the context of why this type exists?

Maybe something on the lines?
```
// Ensures that annotation name property matches its Annotations object key
```